Monika After Story/game/zz_dockingstation.rpy

```diff
@@ -2479,3 +2479,219 @@ label mas_dockstat_found_monika:
+
+#IOSTART LABEL
+#Use this to begin monika file generation
+#REQUIRES THE FOLLOWING GLOBAL VARIABLE TO BE SET TO ALLOW CUSTOM FLOWS
+# - mas_farewells.dockstat_iowait_label
+#   If not set, the generic iowait label is assumed
+label mas_dockstat_iostart:
+    show monika 2dsc
+    python:
+        persistent._mas_dockstat_going_to_leave = True
+        first_pass = True
+
+        # launch I/O thread
+        promise = store.mas_dockstat.monikagen_promise
+        promise.start()
+
+    #Jump to the iowait label
+    if renpy.has_label(mas_farewells.dockstat_iowait_label):
+        jump expression mas_farewells.dockstat_iowait_label
+    #If the one passed in wasn't valid, then we'll use the generic iowait
+    jump mas_dockstat_generic_iowait
+
+
+#GENERIC IOWAIT LABEL
+##NOTE: REQUIRES THE FOLLOWING GLOBAL VARIABLES FOR CUSTOM FLOWS
+# - mas_farewells.dockstat_rtg_label
+# - mas_farewells.dockstat_cancel_dlg_label
+# - mas_farewells.dockstat_wait_menu_label
+#NOTE: If any of these are None (as they default), generic labels are assumed
+label mas_dockstat_generic_iowait:
+    hide screen mas_background_timed_jump
+
+    # we want to display the menu first to give users a chance to quit
+    if first_pass:
+        $ first_pass = False
+        m 1eua "Give me a second to get ready."
+
+        #Prepare the current drink to be removed if needed
+        python:
+            current_drink = MASConsumable._getCurrentDrink()
+            if current_drink and current_drink.portable:
+                current_drink.acs.keep_on_desk = False
+
+        #Get Moni off screen
+        call mas_transition_to_emptydesk
+
+    elif promise.done():
+        # i/o thread is done!
+        #We're ready to go. Let's jump to the rtg label
+        if renpy.has_label(mas_farewells.dockstat_rtg_label):
+            jump expression mas_farewells.dockstat_rtg_label
+        #Otherwise, we don't have a valid label and need to jump to a generic ready to go
+        jump mas_dockstat_generic_rtg
+
+    # display menu options
+    # 4 seconds seems decent enough for waiting.
+    show screen mas_background_timed_jump(4, "mas_dockstat_generic_iowait")
+    menu:
+        m "Give me a second to get ready.{fast}"
+        "Hold on a second!":
+            hide screen mas_background_timed_jump
+            $ persistent._mas_dockstat_cm_wait_count += 1
+
+    # fall thru to the wait wait flow
+    if renpy.has_label(mas_farewells.dockstat_wait_menu_label):
+        call expression mas_farewells.dockstat_wait_menu_label
+    else:
+        #Otherwise fallback to this label
+        call mas_dockstat_generic_wait_label
+
+    #If we returned True, that means we cancelled
+    if _return:
+        #We need to clear all the vars in case we go dockstat again
+        $ mas_farewells.resetDockstatFlowVars()
+
+        #And now, we return what the generic label would have returned, that is if it's a string.
+        if isinstance(_return, unicode):
```

Comment:
Even if in most cases renpy will convert everything into `unicode`, maybe it'd be safer to check for `basestring` just in case?
```suggestion
        if isinstance(_return, basestring):
```